doc/jbuild.rst

```diff
@@ -654,41 +655,54 @@ generated by an OCaml program via:
 
 #. Expansion of lists
 
-Forms that expands to list of items, such as ``${^}``, ``${@}`` or
-``${read-lines:...}`` will always expand to a single string where
-elements are separated by spaces. Inside ``(run <prog> <arguments>)``
-forms you can however split the items as several arguments by
-prefixing the variable with ``!``. Such forms can only be used as a
-whole atom, i.e. they can't be used inside a quoted atom.
-
-For instance in:
+Forms that expands to list of items, such as ``${CC}``, ``${^}``,
+``${@}`` or ``${read-lines:...}``, are suitable to be used in, say,
+``(run <prog> <arguments>)``.  For instance in:
 
 .. code:: scheme
 
     (run foo ${^})
 
-even if there are two dependencies ``a`` and ``b``, the produced
-command will be equivalent to the shell command:
+if there are two dependencies ``a`` and ``b``, the produced command
+will be equivalent to the shell command:
 
 .. code:: shell
 
-    $ foo "a b"
+    $ foo "a" "b"
+
+If you want the two dependencies to be passed as a single argument,
+you have to quote the variable as in:
+
+.. code:: scheme
+
+    (run foo "${^}")
```

Comment:
This is not true at the moment. The double quotes are only seen by the lexer. From the point of view of jbuilder, `hello` and `"hello"` are exactly the same atom.

We could change jbuilder to keep this information and do different things according to whether atoms are quoted or not. This might make sense given that the syntax is different from the user point of view. However, we should have this discussion in a different ticket to avoid delaying this PR.

For now, the user has to add a space at the beginning or end of the atom:

```
(run foo "${^} ")
```

Reply:
Opened https://github.com/janestreet/jbuilder/issues/408